torch/testing/_internal/distributed/rpc/process_group_agent_test_fixture.py

```diff
@@ -13,12 +13,19 @@ def rpc_backend(self):
 
     @property
     def rpc_backend_options(self):
-        return rpc.backend_registry.construct_rpc_backend_options(
-            self.rpc_backend,
-            init_method=self.init_method,
-            # Some tests need additional threads (ex: test_trainer_ps)
-            num_send_recv_threads=8,
-        )
+        try:
+            return self._rpc_backend_options
+        except AttributeError:
```

Comment:
when will this error be triggered?

Reply:
This error actually happens in the default flow when we do not run under single threaded PG agent. When running under single threaded mode, we set `_rpc_backend_options` (see setter method below), when we don't, we call into the below `construct_rpc_backend_options` as before. 

Not sure if there's a better way to write this, but something like the following would also work:
```
if hasattr(self, '_single_threaded_options'):
    return self._single_threaded_options # means we want to run in single-threadd mode
else:
    return construct_rpc_backend_options(...)
```